Source/DiabloUI/text_input.hpp

```diff
@@ -115,7 +115,7 @@ class TextInputState {
 		{
 			const size_t newLength = len_ + text.size();
 			if (newLength > maxLength_) {
-				return TruncateUtf8(text, newLength - maxLength_);
+				return TruncateUtf8(text, maxLength_ - len_);

```

Comment:
We can avoid the `newLength` calculation and the if statement, `TruncateUtf8` does this check.
So simply always `return TruncateUtf8(text, maxLength_ - len_);`